Nuking SequentialExecutor from the in-tree executors (#48528)
mistercrunch/Airflow: airflow-core/docs/core-concepts/executor/index.rst

.. code-block:: bash

    $ airflow config get-value core executor
    LocalExecutor

Executor Types
--------------

There is only one type of executor that runs tasks *locally* (inside the ``scheduler`` process),

> and those that run their tasks *remotely* (usually via a pool of *workers*). Airflow comes configured with the ``LocalExecutor`` by default, which is a local executor, and the simplest option for execution.

Reviewer: There is only one "core" executor that does. But anyone can write an executor using the public interface and many of those could be local. I don't see the reason to specifically call out only one here, I think it's more misleading than it is helpful?

```suggestion
There are two types of executors - those that run tasks *locally* (inside the ``scheduler`` process),
and those that run their tasks *remotely* (usually via a pool of *workers*). Airflow comes configured with the ``LocalExecutor`` by default, which is a local executor, and the simplest option for execution.
```

Author: Just updated it to be more explicit.

> However, the ``LocalExecutor`` is not advised for production use since it runs processes in the scheduler process and that can have an impact on the performance of the scheduler. You can use the ``LocalExecutor``
> for small, single-machine production installations, or one of the remote executors for a multi-machine/cloud installation.

Reviewer 2: That is not correct, LocalExecutor _can_ be used and I even saw multiple schedulers each with a LocalExecutor in the wild.
```suggestion
However, as the ``LocalExecutor`` runs processes in the scheduler process that can have an impact on the performance of the scheduler. You can use the ``LocalExecutor``
for small, single-machine production installations, or one of the remote executors for a multi-machine/cloud installation.
```

Author: Yeah more accurate!
